### DISCUSSION TITLE
Get the redirect url

### DISCUSSION THREAD

**Discussion by jrean** [wavezync/vue3-google-signin#43] [Q&A]
Hi,
Using the `useCodeClient` composable how can I get access to the redirect url value that was used? I need to provide that exact value to my backend for code validation. Whatever value I try for my backend validation I receive a `redirect_uri_mismatch`.
I see in the [source code](https://github.com/syetalabs/vue3-google-signin/blob/ede9fc2ed871c5d745e25807bb0bdf7c1bc9ae5b/src/composables/useCodeClient.ts#L148) it returns `codeRequestRedirectUrl` but it didn't help.

Thxs in advance

**Comment by kasvith** [comment]
Please follow this https://vue3-google-signin.syetalabs.io/guide/creating-custom-buttons.html#verify-code-on-the-backend

**Reply by jrean** [reply]
Thank you. I finally found out I should be able to provide (or not) `redirect_uri` part of the options:
https://github.com/syetalabs/vue3-google-signin/blob/main/src/utils/oauth2.ts#L92
```
if (options.redirect_uri === undefined) {
    queryParams.append("redirect_uri", window.origin);
  } else {
    queryParams.append("redirect_uri", options.redirect_uri);
  }
```

Since I was not manually providing it, it was deferring to `window.origin` (http://domain.com) while on my backend I was expecting the full url (http://domain.com/login). So it works properly as long as I keep the value of `window.origin`.

But if I try to set a custom option `redirect_uri` value:

```
const { isReady, login } = useCodeClient({
  redirect_uri: 'http://domain.com/login',

  onSuccess: handleGoogleLoginSuccess,
  onError: handleGoogleLoginError,
  // other options
})
```

Then on the backend I receive a `redirect_uri_mismatch` while providing `http://domain.com/login` as redirect_uri value...
So by default it works but if I want to use a custom redirect_uri value it doesn't work.

- It would be nice to have a simple note in the official documentation about this default behaviour to use the `window.origin` value.
- Any idea why it stops working as soon as I set a custom `redirect_uri` value?

**Reply by kasvith** [reply]
@jrean second point seems like a bug
